src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -108,6 +108,36 @@ impl ConvertShares {
     }
 }
 
+pub async fn convert_all_shares<F: Field>(
+    ctx: &ProtocolContext<'_, F>,
+    input: &[u64],
+) -> Result<Vec<Vec<Replicated<F>>>, BoxError> {
+    let converted_shares = try_join_all(zip(repeat(ctx), input).enumerate().map(
+        |(record_id, (ctx, row))| async move {
+            ConvertShares::new(XorShares {
+                num_bits: 64,
+                packed_bits: *row,
+            })
+            .execute(
+                ctx.narrow(&ModulusConversion(record_id.try_into().unwrap())),
```

Comment:
This is definitely wrong. This enum only supports values up to 64... we should be able to sort millions.

I *think* there is this hack where if you have NEVER called `narrow()` then everything just works. Like, you can clone the context and invoke the same thing multiple times for the different record IDs. You can use this trick exactly once though. As soon as you've called "narrow()" it doesn't work anymore.

Reply:
Doesnt this get called per bit? Then we can run Modulus conversion for upto 64 bits?

Reply:
```
ConvertShares::new(XorShares {
    num_bits: 64,
    packed_bits: row,
}).execute(ctx, record_id).await?
```

That will convert one input `row` into a `Vec` of `Replicated<F>` having length `64`. It converts all the bits at once with one invocation of this protocol.